### ISSUE TITLE
"Using SSL in Twisted" should use twisted.internet.ssl.CertificateOptions

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6273]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)| @hynek reported|
|-|-|
|Trac ID|trac#6273|
|Type|enhancement|
|Created|2013-01-22 13:45:32Z|
|Branch|https://github.com/twisted/twisted/tree/ssl-contextfactory-docs-6273-2|

Since `twisted.internet.ssl.CertificateOptions` is supposed to supersede `twisted.internet.ssl.DefaultOpenSSLContextFactory`, there should be tutorial material available instead of encouraging people using the "old way".

<details><summary>Searchable metadata</summary>

```
trac-id__6273 6273
type__enhancement enhancement
reporter__hynek hynek
priority__normal normal
milestone__ 
branch__branches_ssl_contextfactory_docs_6273_2 branches/ssl-contextfactory-docs-6273-2
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1358862332000000 1358862332000000
changetime__1363644424000000 1363644424000000
version__None None
owner__exarkun exarkun
cc__hynek
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [36969]) Branching to 'ssl-contextfactory-docs-6273'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [37494]) Branching to 'ssl-contextfactory-docs-6273-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

The ssl howto is now updated.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/ssl-contextfactory-docs-6273-2).

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_6"></a>|@jonathanj set owner to @exarkun|
|-|-|

Looks mostly good, I have a few small points:

 1. The code listing in the "startTLS server" section imports OpenSSL but doesn't seem to use it, are there side-effects to importing OpenSSL?

 2. Perhaps the code listings could benefit from using `twisted.internet.task.react`.

Other than that the information looks factual and I would say it is clearer now; looks good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Thanks for the review!

> Perhaps the code listings could benefit from using twisted.internet.task.react.

Probably so.  There are no Deferreds around to easily make the transition, though, so I'd rather leave this for another time.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [37632]) Merge ssl-contextfactory-docs-6273-2

Author: exarkun
Reviewer: jonathanj
Fixes: [#6273](https://github.com/twisted/twisted/issues/6273)

Switch the SSL howto and SSL examples from the older, less featureful SSL context factory APIs
to the newer, preferred `CertificateOptions` API.
